File: service1.py
import os
import subprocess
import requests
from flask import Flask, jsonify

app = Flask(__name__)

# Function to get system info
def get_system_info():
    ip_address = subprocess.getoutput("hostname -I").strip()
    processes = subprocess.getoutput("ps -ax")
    disk_space = subprocess.getoutput("df -h /")
    uptime = subprocess.getoutput("uptime -p")
    return {
        "IP Address": ip_address,
        "Processes": processes,
        "Disk Space": disk_space,
        "Uptime": uptime
    }

@app.route('/')
def index():
    # Gather information for Service1
    service1_info = get_system_info()

    # Communicate with Service2 to gather its info
    try:
        response = requests.get("http://service2:5000/")
        service2_info = response.json()
    except requests.exceptions.RequestException as e:
        service2_info = {"error": "Service2 not reachable"}

    # Return combined info from Service1 and Service2
    return jsonify({
        "Service1": service1_info,
        "Service2": service2_info
    })

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=8199)

File: service1.js
const express = require('express');
const { execSync } = require('child_process');
const axios = require('axios');

const app = express();
const PORT = 8199;

let isSleeping = false;

function getSystemInfo() {
    const ipAddress = execSync('hostname -I').toString().trim();
    const processes = execSync('ps -ax').toString();
    const diskSpace = execSync('df -h /').toString();
    const uptime = execSync('uptime -p').toString().trim();

    return {
        "IP Address": ipAddress,
        "Processes": processes,
        "Disk Space": diskSpace,
        "Uptime": uptime
    };
}

app.get('/request', async (req, res) => {
    if (isSleeping) {
        return res.status(503).json({ message: "Service1 is sleeping" });
    }

    const service1Info = getSystemInfo();

    let service2Info;
    try {
        const response = await axios.get('http://service2:5000/');
        service2Info = response.data;
    } catch (error) {
        service2Info = { error: "Service2 not reachable" };
    }

    res.json({
        "Service1": service1Info,
        "Service2": service2Info
    });

    isSleeping = true;
    setTimeout(() => {
        isSleeping = false;
    }, 2000);
});

app.post('/stop', (req, res) => {
    res.send('Stopping services...');
    setTimeout(() => {
        process.exit(0);
    }, 2000);
});

app.listen(PORT, () => {
    console.log(`Service1 running on port ${PORT}`);
});
